
<!DOCTYPE html>
<!--
  Forty by HTML5 UP
  html5up.net | @ajlkn
  Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>

<head>
  <title>MTN Region SARS-CoV-2 Training | StaPH-B</title>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
  <!--[if lte IE 8]><script src="/assets/js/ie/html5shiv.js"></script><![endif]-->
  <link rel="stylesheet" href="/assets/css/main.css" />
  <script src="https://kit.fontawesome.com/b021ccefd1.js" crossorigin="anonymous"></script>
  <!--[if lte IE 9]><link rel="stylesheet" href="/assets/css/ie9.css" /><![endif]-->
  <!--[if lte IE 8]><link rel="stylesheet" href="/assets/css/ie8.css" /><![endif]-->
</head>


<body>

    <!-- Wrapper -->
<div id="wrapper">

<!-- Header -->
<header id="header">
  <a href="/" class="logo"><strong>StaPH-B</strong> <span>State Public Health Bioinformatics Group</span></a>
  <nav>
    <a href="#menu">Menu</a>
  </nav>
</header>

<!-- Menu -->
<nav id="menu">
  <ul class="links">
        
        
    
        
    
        
            <li><a href="/">Home</a></li>
        
    
        
    
        
    
        
    
        
    
    
        
    
        
            <li><a href="/about.html">About StaPH-B</a></li>
        
    
        
    
        
            <li><a href="/resources.html">Resources</a></li>
        
    
        
            <li><a href="/training.html">Training</a></li>
        
    
        
            <li><a href="/videos.html">Videos</a></li>
        
    
  </ul>
</nav>



<!-- Main -->

<div id="main" class="readable">

<!-- One -->
<section id="one">
  <div class="inner">
    <header class="major">
      <h1>MTN Region SARS-CoV-2 Training</h1>
      
      <h4>By: Joel Sevinsky</h4>
      
      <h4>June 26, 2020</h4>
    </header>
    
    <h3 id="training-prerequisites">Training Prerequisites</h3>
<p>In order for this training to be effective for all students participating, each student should perform the following before the first class:</p>
<ol>
  <li>Confirm that they can connect to their VM.</li>
  <li>Watch a 75 minute video on the linux operating system.</li>
  <li>Watch a short video on tmux, a linux application that allows for multiple terminals.</li>
</ol>

<h6 id="connect-to-your-vm"><em>Connect to your VM</em></h6>

<p>In the slack channel will be listed your VM IP address, along with a ssh key file. This information will be posted on Monday, June 29th. In order to connect to this VM, you will need to download the terminal program MobaXterm. MobaXterm can be downloaded from <a href="https://mobaxterm.mobatek.net/download-home-edition.html">https://mobaxterm.mobatek.net/download-home-edition.html</a>. IMPORTANT: Please download the “Portable edition”, not the “Installer edition”. The “Portable edition” will allow you to run the application without installing it, thus no admin rights needed. Once you have downloaded MobaXterm and have access to your IP address and ssh key, watch the youtube video on <a href="https://youtu.be/CWEA86aRkg8">How to Connect to Your VM</a>. I will also post the username and passphrase in slack.</p>

<h6 id="linux-video"><em>Linux video</em></h6>

<p>The video link is <a href="https://www.youtube.com/watch?v=oxuRxtrO2Ag">https://www.youtube.com/watch?v=oxuRxtrO2Ag</a>. This video will give you a basic understanding of navigating around the bash terminal in linux. It is well worth the 75 minutes before the course, working within your own VM.</p>

<h6 id="tmux-video"><em>Tmux video</em></h6>

<p>Many of the processes we will be running will require long processing times that could be interrupted by network connection issues. To avoid this, we will perform the exercises in the class using the tmux application. This <a href="https://youtu.be/BHhA_ZKjyxo">tmux tutorial video</a> will guide you through the process of using tmux.</p>

<p>Some additional resources that you will find useful when starting out are:</p>

<ul>
  <li><a href="https://fossbytes.com/a-z-list-linux-command-line-reference/">Linux command reference</a></li>
  <li><a href="https://tmuxcheatsheet.com/">tmux cheat sheet</a></li>
</ul>

<h3 id="agenda-overview">Agenda Overview</h3>

<p><em><strong>Session 1 - Linux</strong></em> - Monday July 6th, 12 PM and 2 PM MDT <br />
<em><strong>Session 2 - Background on Viral Genomics and Coronaviruses</strong></em>  - Tuesday, July 7th, 12 PM and 2 PM MDT <br />
<em><strong>Session 3 - Sequencing Methods for SARS-CoV-2</strong></em> - Wednesday July 8th, 12 PM and 2 PM MDT <br />
<em><strong>Session 4 - StaPH-B Toolkit</strong></em> - Monday July 13th, 12 PM and 2 PM MDT <br />
<em><strong>Session 5 - UPHL BioNGS</strong></em> - Tuesday July 14th, 12 PM and 2 PM MDT <br />
<em><strong>Session 6 - Commercial Options</strong></em> - Wednesday July 15th, 12 PM and 2 PM MDT <br />
<em><strong>Session 7 - NGS Data Visualization for QC of Results</strong></em> - Monday July 20th, 12 PM and 2 PM MDT <br />
<em><strong>Session 8 - Data Sharing with GISAID and NCBI</strong></em> - Tuesday July 21st, 12 PM and 2 PM MDT <br />
<em><strong>Session 9 - Data Visualization in Nextstrain</strong></em> - Wednesday July 22nd, 12 PM and 2 PM MDT</p>

<p>Office hours will be offered each week on Thursday from 2 PM to 4 PM MDT, on Friday from 9 AM to 11 AM MDT, and by request.</p>

<h3 id="expanded-agenda-will-contain-links-videos-code-as-course-progresses">Expanded Agenda (will contain links, videos, code, as course progresses)</h3>
<h5 id="week-1">Week 1</h5>

<p><em><strong>Session 1 - Linux</strong></em></p>

<p><strong><a href="https://youtu.be/R_SR7A8o6d8">Recording for Session 1 - 12:00 PM MDT</a></strong></p>

<p><strong><a href="https://youtu.be/XbU_UmFSkf8">Recording for Session 1 - 3:00 PM MDT</a></strong></p>

<ul>
  <li>Linux basics</li>
  <li>Working in Tmux
    <ul>
      <li><a href="https://tmuxcheatsheet.com/">Tmux Cheat Sheet</a></li>
      <li><a href="https://www.youtube.com/watch?v=oxuRxtrO2Ag">Tmux Video</a></li>
    </ul>
  </li>
  <li>Connecting to Basespace
    <ul>
      <li><a href="https://developer.basespace.illumina.com/docs/content/documentation/cli/cli-overview">Basespace CLI</a></li>
    </ul>
  </li>
  <li>Connecting to cloud resources
    <ul>
      <li><a href="https://cloud.google.com/storage/docs/gsutil">gsutil</a></li>
    </ul>
  </li>
  <li>Transferring data to your VM for the class</li>
</ul>

<p><strong>Commands used in this session</strong></p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>***bash commands***

dir                           #lists out a directory
ls                            #also lists out a directory
ls -la                        #lists out a directroy with details
ll                            #alias for ls -la
cd &lt;directory&gt;                #change directory
cd                            #will bring you to your home directory
mkdir &lt;dir&gt;                   #make directory
rmdir &lt;dir&gt;                   #remove directory if it is empty
rm -rf &lt;dir&gt;                  #remove directory if it is full
gzip &lt;name.fastq&gt;             #zip a read file

***tmux***

tmux ls                       #list open tmux sessions
tmux new -s &lt;name&gt;            #start a tmux session called &lt;name&gt;
tmux a -t &lt;name&gt;              #attach to tmux session called &lt;name&gt;
tmux kill-session -t &lt;name&gt;   #kill tmux session called &lt;name&gt;
Ctrl-b ,                      #rename current window
Ctrl-b c                      #create new window
Ctrl-b n                      #go to next window
tmux info                     #help

***basespace cli***

bs auth                                   #authenticate to Basespace
bs list projects                          #list projects in basespace
bs download project -n &lt;name&gt; -o &lt;dir&gt;    #download project files to &lt;dir&gt;

***gsutils***

gsutil ls                                 #list storage buckets
gsutil cp &lt;source&gt; &lt;destination&gt;          #cp data from storage bucket

***sra-toolkit*** - might need to "sudo apt-get install sra-toolkit"

prefetch &lt;sra_id&gt;
fastq-dump --split-files --gzip &lt;sra_id&gt;
</code></pre></div></div>

<p><em><strong>Session 2 - Background on Viral Genomics and Coronaviruses</strong></em></p>

<p><strong><a href="https://youtu.be/_UoIDdXmI2A">Recording for Session 2 - 12 PM MDT</a></strong></p>

<p><strong><a href="https://youtu.be/YqAFqWWMDUM">Recording for Session 2 - 3 PM MDT</a></strong></p>

<p><strong><a href="/assets/trainings_assets/MTN-2020-Session2-CoronavirusGenomics.pdf">Session 2 Slides</a></strong></p>

<ul>
  <li>Viral genomics primer</li>
  <li>Considerations of bacteria vs viral pathogens</li>
  <li>Coronavirus, the new flu.</li>
  <li><a href="https://paperpile.com/shared/dU0ZnG">Papers on Coronaviruses</a></li>
</ul>

<p><strong>REMINDER 7/7</strong> –&gt; register for a <a href="https://www.gisaid.org/registration/register/">GISAID account</a></p>

<p><em><strong>Session 3 - <a href="https://github.com/CDCgov/SARS-CoV-2_Sequencing">Sequencing Methods for SARS-CoV-2</a></strong></em></p>

<p><strong><a href="https://youtu.be/vwqvZcyLD_E">Recording for Session 3 - 12 PM MDT</a></strong></p>

<p><strong><a href="https://youtu.be/cXHPWbMGR8g">Recording for Session 3 - 3 PM MDT</a></strong></p>

<ul>
  <li>Metagenomics, enrichment, amplicon</li>
  <li>The <a href="https://artic.network/ncov-2019">ARTIC</a> protocol</li>
  <li><a href="https://www.protocols.io/file-manager/E2F61524120340C3B7C07A7C9E755CB0">ARTIC + Illumina DNA Flex</a> at protocols.io.</li>
</ul>

<h5 id="week-2">Week 2</h5>

<p><em><strong>Session 4 - <a href="https://github.com/StaPH-B/staphb_toolkit">StaPH-B Toolkit</a></strong></em></p>

<p><strong><a href="https://youtu.be/h6l7tdNWXU4">Recording for Session 4 - 12 PM MDT</a></strong> (Watch the 3 PM instead)</p>

<p><strong><a href="https://youtu.be/nrXgNtbghCs">Recording for Session 4 - 3 PM MDT</a></strong> (This is the better recording)</p>

<ul>
  <li>Monroe pipeline</li>
  <li>pe_assembly</li>
  <li>cluster_analysis</li>
</ul>

<p><a href="https://staph-b.github.io/staphb_toolkit/workflow_docs/monroe/">Documentation on Monroe</a></p>

<p>Some commands to make today’s session easier</p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>tmux new -s session4

mkdir session4-1
cd session4-1
gsutil ls
gsutil ls gs://mtn-reads/
gsutil -m cp -r gs://mtn-reads/session4/* .
tar -xf reads.tar
mkdir reads
mv *.gz reads
rm reads.tar
staphb-wf monroe
staphb-wf monroe pe_assembly
staphb-wf monroe pe_assembly --primers V3 --output pe_assembly_1 --config 20-07-11_pe_assembly.config reads

staphb-wf monroe cluster_analysis
staphb-wf monroe cluster_analysis --output cluster_analysis_1 --config 20-07-11_cluster_analysis.config pe_assembly_1/assemblies/


</code></pre></div></div>

<p><em><strong>Session 5 - UPHL BioNGS</strong></em></p>
<ul>
  <li><a href="https://github.com/UPHL-BioNGS/Cecret">Cecret SARS-CoV-2 pipeline</a></li>
</ul>

<p><strong><a href="https://youtu.be/jWEyBGTq6W8">Recording for Session 5 - 12 PM MDT</a></strong></p>

<p><strong><a href="https://youtu.be/Z-QL8s1BBK4">Recording for Session 5 - 3 PM MDT</a></strong></p>

<p>Data files</p>

<p><a href="https://storage.googleapis.com/staphb-resources/session5_files.tar">session5_files.tar</a></p>

<p><a href="https://storage.googleapis.com/staphb-resources/session5_files_2.tar">session5_files_2.tar</a></p>

<p>Some commands to make today’s exercises easier</p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#see if you have a tmux session open
tmux ls

#create a new tmux session called session5
tmux new -s session5

#create new window in your tmux session
&lt;create new window in tmux with Ctrl-b, c&gt;

#copy over session files
gsutil -m cp -r gs://mtn-reads/session5/* .

#extract the session files
tar -xf session5_files_2.tar

#change dir into cecret_session5
cd cecret_session5

#check out covid_samples.txt
column -t covid_samples.txt | less -S

#copy new Cecret.nf to your Cecret folder
cp Cecret.nf ~/Cecret/

#check thatt Cecret.nf was updated
ll ~/Cecret

#launch Cecret - will take about 50 minutes with 10 specimens
~/nextflow run ~/Cecret/Cecret.nf -c cecret.docker.nextflow.google.config

#go up one directory to session5
cd ..

#unpack bakeshow data
tar -xf session5_files.tar

#move into bakeshow directory
cd cecret_bakeshow

#check out run_results.txt
column -t run_results.txt | less -S


</code></pre></div></div>

<p><em><strong>Session 6 - Cecret and Monroe Continued</strong></em></p>

<p><em><strong>UPDATE:</strong></em> Most likely we will continue looking at Monroe and Cecret results</p>

<p><strong><a href="https://youtu.be/RJecZ8vA1S8">Recording for Session 6 - 12 PM MDT</a></strong></p>

<p><strong><a href="https://youtu.be/fwwtr2vYgaE">Recording for Session 6 - 3 PM MDT</a></strong></p>

<p>Not covered:</p>
<ul>
  <li><a href="https://go.qiagen.com/QDI-COVID19">CLC Genomics Workbench</a></li>
</ul>

<h5 id="week-3">Week 3</h5>

<p><em><strong>Session 7 - Data Sharing with GISAID and NCBI</strong></em></p>

<p><strong><a href="https://youtu.be/GgoQjLxYouA">Recording for Session 7 - 12 PM MDT</a></strong></p>

<p><strong><a href="https://youtu.be/comk7G9ddvg">Recording for Session 7 - 3 PM MDT</a></strong></p>

<p>Topics update:</p>
<ul>
  <li>Data sharing with <a href="https://www.gisaid.org/">GISAID</a></li>
</ul>

<p>Topics not covered, future webinars:</p>
<ul>
  <li><a href="https://go.qiagen.com/QDI-COVID19">CLC Genomics Workbench</a></li>
  <li><a href="https://www.megasoftware.net/">Mega-X</a></li>
  <li><a href="http://software.broadinstitute.org/software/igv/">Integrated Genomics Viewer</a> (IGV)</li>
</ul>

<p><em><strong>Session 8 - Data Sharing with GISAID and NCBI</strong></em></p>

<p><strong><a href="https://youtu.be/5S2lFWIloso">Recording for Session 8 - 12 PM MDT</a></strong></p>

<p><strong><a href="https://youtu.be/qyomji9WpxY">Recording for Session 8 - 3 PM MDT</a></strong></p>

<p>Topics update:</p>
<ul>
  <li>Data sharing with <a href="https://www.ncbi.nlm.nih.gov/sra">SRA</a></li>
</ul>

<p><em><strong>Session 9 - More Data Sharing</strong></em></p>

<p><strong><a href="https://youtu.be/lTejITeZgt0">Recording for Session 9 - 12 PM MDT</a></strong></p>

<p><strong><a href="https://youtu.be/tlUXLaP2au4">Recording for Session 9 - 3 PM  MDT</a></strong></p>

<p>Topics update:</p>
<ul>
  <li>Data sharing with <a href="https://www.ncbi.nlm.nih.gov/genbank/">Genbank</a></li>
</ul>

<p>Topics not covered below, will have future webinars.</p>
<ul>
  <li>Intro to <a href="https://nextstrain.org/">NextStrain</a> for visualization</li>
  <li>Running on <a href="https://auspice-us.herokuapp.com/">herokuapp.com</a></li>
  <li><a href="https://nextstrain.github.io/ncov/">Tutorial</a></li>
  <li><a href="https://clades.nextstrain.org/">NextStrain Clades</a></li>
</ul>

  </div>
</section>

</div>

    <!-- Footer -->
  <footer id="footer">
    <div class="inner">
      <ul class="icons">
        
          
            <li>
              <a href="https://github.com/StaPH-B" class="icon alt fa-github" target="_blank" rel="noopener noreferrer" aria-label="GitHub">
                <span class="label">GitHub</span>
              </a>
            </li>
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
            <li>
              <a href="https://join.slack.com/t/staph-b-dev/shared_invite/zt-w4ivhtq9-2XypNGWXY9vmyeWf0lABng" class="icon alt fa-slack" target="_blank" rel="noopener noreferrer" aria-label="Slack">
                <span class="label">Slack</span>
              </a>
            </li>
          
        
          
            <li>
              <a href="https://www.youtube.com/channel/UC4GughelEJxkJORLTCd51eg" class="icon alt fa-youtube" target="_blank" rel="noopener noreferrer" aria-label="youtube">
                <span class="label">youtube</span>
              </a>
            </li>
          
        
      </ul>
      <ul class="copyright">
        <li>&copy; StaPH-B State Public Health Bioinformatics Group</li>
        <li>Built from the <a href="https://github.com/andrewbanchich/forty-jekyll-theme" target="_blank">forty-jekyll-theme</a></li>
      </ul>
    </div>
  </footer>

</div>

<!-- Scripts -->
  <script src="/assets/js/jquery.min.js"></script>
  <script src="/assets/js/jquery.scrolly.min.js"></script>
  <script src="/assets/js/jquery.scrollex.min.js"></script>
  <script src="/assets/js/skel.min.js"></script>
  <script src="/assets/js/util.js"></script>
  <!--[if lte IE 8]><script src="/assets/js/ie/respond.min.js"></script><![endif]-->
  <script src="/assets/js/main.js"></script>


</body>

</html>
